### PR TITLE
Execution with only pipe redirections

### LINKED ISSUE #21: Implement execution without redirections
As a first step, we should implement the execution without redirections.
The execution shall check whether the user input is a pipeline and for each command to run, check if it is a builtin.
In the case of a pipeline, no matter how long it is, no more than 3 file descriptors should be opened at the same time, in addition to the 3 standard file descriptors.

### PR DESCRIPTION
**Pull request by JonathanDUFOUR**
The made changes are related to the issue https://github.com/Saphirac/minishell/issues/21.
It is now possible to run commands/builtins directly from user input.
Only the pipe redirections have been implemented. The input redirections from a file/heredoc and the output redirections to a file shall be implemented later.

### CODE REVIEW

File: src/exec/pid/lst/pid_lst_wait.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	char const *const	str;
+};
+
+static t_signaled_str const		g_signaled_str[] = {
+{SIGQUIT, "Quit (core dumped)"},
+{SIGSEGV, "Segmentation fault (core dumped)"},
+{SIGINT, ""},
+{0}
+};
+
```
Comment (Saphirac) [inline src/exec/pid/lst/pid_lst_wait.c:29]: No comment ?

File: src/prompt.c [hunk 2]
```diff
@@ -15,11 +15,12 @@
...
 {
 	print_tokens(&shell->tokens);
-	printf("pipeline : %d\n", shell->is_pipeline);
+	if (execution(shell))
+		exit(EXIT_FAILURE);
 }
 
 /**
- * @brief	Display the prompt and get the user input.
+ * @brief	Display the prompt and get the user inpuredirectionst.
```
Comment (Saphirac) [inline src/prompt.c]: Typo

File: src/tokens/lst_tokens/token_lst_del_one.c [hunk 3]
```diff
@@ -36,5 +36,5 @@ void	token_lst_del_one(t_token_lst *const list, t_token *const node)
 	}
 	--list->size;
 	free(node->str);
-	free(node);
+	free((void *)node);
```
Comment (Saphirac) [inline src/tokens/lst_tokens/token_lst_del_one.c:39]: Not convinced.

**Review by Saphirac — CHANGES_REQUESTED** [review]
Please just check the comments made !

File: src/tokens/lst_tokens/token_lst_del_one.c (hunk 3 above)
Comment (JonathanDUFOUR) [reply src/tokens/lst_tokens/token_lst_del_one.c:39]: After some investigations, I believe that it is a better idea to keep it const and cast it as non-const only when calling `free()` (cf https://stackoverflow.com/questions/2819535/unable-to-free-const-pointers-in-c#comment2860291_2819594)

**Review by Saphirac — APPROVED** [review]
I accept the proof you gave me for the free.

**Pull request merged**